### DISCUSSION TITLE
Example with Javascript only (no typescript)

### DISCUSSION THREAD

**Discussion by laurent512** [TanStack/table#4096] [Q&A]
Hi, 

I am using Next.js using only javascript , no typescript.
In the currently available I found only ts examples, which I find confusing ( being unfamiliar with ts) 

Is there any example showing tanstack table v8 with javascript only (with react) ?

Thank you

**Comment by tannerlinsley** [comment]
The examples would be exactly the same, just without any type syntax.

**Reply by o-t-w** [reply]
It would be really useful if there were some getting started pages that showed simple examples. And for people totally unfamiliar with TypeScript it is difficult to know which parts are type syntax that they should ignore. 

I am looking at the Basic example on CodeSandbox and trying to read the docs but i have no idea what is going on. 

There are just so many questions not really addressed in the docs. What is a flexRender? Why is it even called that? Do you need to call getHeaderGroups if you haven't grouped your headings? Why do some accessors use a string while others use `columnHelper.accessor(row => row.lastName`?

**Reply by endrureza** [reply]
@o-t-w you could find it here https://stackoverflow.com/questions/73885732/how-to-implement-react-table-without-typescript, someone gave an example without typescript

**Comment by jlomoglio** [comment]
I agree, we opted to stick with v7, the new version lacks alot of stuff. While a lot of developers my use Typescript, a lot do not. It's a shame that the docs suck and there is not very much out there on YouTube to get a good tutorial. So I would suggest using v7 or try the MUI table, untill v8 becomes mature enough

**Comment by shuvro111** [comment]
@laurent512 @o-t-w I found this [example](https://github.com/OneLightWebDev/react-table-tutorial) with javascript and this [YouTube video](https://www.youtube.com/watch?v=CjqG277Hmgg) explains it. You can also take a look at this [codesandbox](https://codesandbox.io/p/github/shuvro111/react-table-tutorial/main?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt2vp7e300063b6hdewxlrox%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt2vp7e300023b6h14kh17by%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt2vp7e300043b6hpyf2n9oh%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt2vp7e300053b6h99di7zqi%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt2vp7e300023b6h14kh17by%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt2vp7e200013b6hze2pon4t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clt2vp7e300023b6h14kh17by%2522%252C%2522activeTabId%2522%253A%2522clt2vp7e200013b6hze2pon4t%2522%257D%252C%2522clt2vp7e300053b6h99di7zqi%2522%253A%257B%2522id%2522%253A%2522clt2vp7e300053b6h99di7zqi%2522%252C%2522activeTabId%2522%253A%2522clt2vrczp00he3b6hd9kx39mz%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522id%2522%253A%2522clt2vrczp00he3b6hd9kx39mz%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clt2vp7e300043b6hpyf2n9oh%2522%253A%257B%2522id%2522%253A%2522clt2vp7e300043b6hpyf2n9oh%2522%252C%2522activeTabId%2522%253A%2522clt2vpa7i00453b6hz6w2075m%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt2vp7e300033b6h4g3dyb6f%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clt2vp96x0011dlijabwceomh%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clt2vpa7i00453b6hz6w2075m%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522CSB_RUN_OUTSIDE_CONTAINER%253D1%2520devcontainer%2520templates%2520apply%2520--template-id%2520%255C%2522ghcr.io%252Fdevcontainers%252Ftemplates%252Fjavascript-node%255C%2522%2520--template-args%2520%27%257B%257D%27%2520--features%2520%27%255B%255D%27%2522%252C%2522id%2522%253A%2522clt2vprcf005o3b6hqc3vokea%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) link to see it in action.
